### PR TITLE
publish machine json

### PR DESCRIPTION
**Pull request by pq**
Follow-up from #2359, adds machine json to the published docs.

See: https://dart-lang.github.io/linter/lints/machine/rules.json

Ultimate format and desired metadata still TBD.

(If desired we could pull this into `tools_metadata` as well but doing the work here feels right.)

/fyi @csells

/cc @devoncarew

### CODE REVIEW

File: tool/doc.dart [hunk 1]
```diff
@@ -61,14 +62,26 @@ These rules are under active development.  Feedback is
 
 const ruleLeadMatter = 'Rules are organized into familiar rule groups.';
 
+final effectiveDartRules = <String>[];
```
Comment (pq) [inline tool/doc.dart:65]: Sorry for all the churn.  The file was unsorted but should have been.

File: tool/doc.dart [hunk 2]
```diff
@@ -422,6 +321,22 @@ class RuleHtmlGenerator {
 ''';
 }
 
+class MachineSummaryGenerator {
```
Comment (pq) [inline tool/doc.dart:324]: This is the only interestingly new bit.

**Review by devoncarew — APPROVED** [review]

File: tool/doc.dart [hunk 3]
```diff
@@ -422,6 +321,22 @@ class RuleHtmlGenerator {
...
+  final Iterable<LintRule> rules;
+
+  MachineSummaryGenerator(this.rules);
+  void generate(String filePath) {
```
[the remark this answers is not in the document]
Comment (pq) [reply tool/doc.dart:328]: yep. We enforce this actually in the build (see: `validate_format_test.dart).  Does look odd without a blank line I agree though...

[the remark this answers is not in the document]
Comment (pq) [reply tool/doc.dart:329]: (line added)

**Pull request merged**